src/operators/mpo.jl

```diff
@@ -55,6 +55,10 @@ DenseMPO(mpo::MPO) = mpo isa DenseMPO ? copy(mpo) : MPO(map(TensorMap, parent(mp
 Base.parent(mpo::MPO) = mpo.O
 Base.copy(mpo::MPO) = MPO(map(copy, mpo))
```

Comment:
```suggestion
Base.copy(mpo::MPO) = MPO(copy.(parent(mpo)))
```

I think this would also fix the issue actually: the problem is that you want to preserve the container type, which actually does happen with broadcasting.

This entire thing is caused by the annoying fact that mpo's are abstractvector, and similar gives you a regular vector by default, thereby losing the `Infinite` information. In hindsight these really should not be `AbstractVector` to begin with, but oh well...